plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogParser.java

```diff
@@ -167,7 +168,14 @@ public static Object deserializePartitionValue(DeltaLakeColumnHandle column, Opt
 
     private static Long readPartitionTimestamp(String timestamp)
     {
-        LocalDateTime localDateTime = LocalDateTime.parse(timestamp, PARTITION_TIMESTAMP_FORMATTER);
+        LocalDateTime localDateTime;
+        try {
+            localDateTime = LocalDateTime.parse(timestamp, PARTITION_TIMESTAMP_FORMATTER);
+        }
+        catch (DateTimeParseException ignored) {
+            // type widening date->timestamp has occurred and this value is of date format
+            localDateTime = LocalDate.parse(timestamp).atTime(LocalTime.MIN);
+        }
```

Comment:
Ideally, we should avoid this exception-driven logic. Could you leave a TODO comment?

Reply:
Yes, something like appending another pattern?